src/CommunityToolkit.Maui.MediaElement/MediaSource/MediaSource.shared.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.ComponentModel;
+using CommunityToolkit.Maui.MediaElement.Converters;
+using Microsoft.Maui.Controls.Internals;
+
+namespace CommunityToolkit.Maui.MediaElement;
+
+/// <summary>
+/// Represents a source that can be played by <see cref="MediaElement"/>.
+/// </summary>
+[TypeConverter(typeof(MediaSourceConverter))]
+public abstract class MediaSource : Element
+{
+	readonly WeakEventManager weakEventManager = new();
+
+	internal event EventHandler SourceChanged
+	{
+		add => weakEventManager.AddEventHandler(value);
+		remove => weakEventManager.RemoveEventHandler(value);
+	}
+
+	/// <summary>
+	/// An implicit operator to convert a string value into a <see cref="MediaSource"/>.
+	/// </summary>
+	/// <param name="source">Full path to a local file (starting with <c>file://</c>) or an absolute URI.</param>
+	[Preserve(Conditional = true)]
```

Comment:
Not sure if this has an effect on the `One .NET` world since this is from Mono.
Anyway, since our lib isn't linker-safe we can remove it

Reply:
```suggestion
```